frontend/src/actions/receive/abandon_action.ts

```diff
@@ -8,7 +8,8 @@ class AbandonAction {
     }
 
     public Invoke() {
-        this.game.setAbandoned(true)
+        this.game.setEndGameReason('abandon')
```

Comment:
'abandon' should be a constant variable instead of a "magic string" copy-pasted anywhere needed, since this copy-paste strategy  may result in typos and errors difficult to find.

Reply:
done